compiler/rustc_lint/src/unused.rs

```diff
@@ -1349,9 +1349,8 @@ declare_lint! {
     /// ### Example
     ///
     /// ```rust
-    /// #![feature(box_syntax)]
     /// fn main() {
-    ///     let a = (box [1, 2, 3]).len();
+    ///     let a = (Box::new([1, 2, 3])).len();
```

Comment:
```suggestion
    ///     let a = (#[rustc_box] Box::new([1, 2, 3])).len();
```
This lint only triggers when a box expression is used at HIR level.